slang.h

```diff
@@ -4301,6 +4307,14 @@ namespace slang
             ITypeConformance** outConformance,
             SlangInt conformanceIdOverride,
             ISlangBlob** outDiagnostics) = 0;
+
+            /** Computes the hash for the linkage, which includes preprocessor defines, the compiler version,
+                and other compiler options. This is then merged with the hash produced for the program to
+                produce a key that can be used with the shader cache.
+            */
+        virtual SLANG_NO_THROW SlangResult SLANG_MCALL computeDependencyBasedHash(
+            SlangInt targetIndex,
+            Checksum* outHash) = 0;
```

Comment:
Probably doesn't, `RendererBase::getEntryPointCodeFromShaderCache()` should be the only location it's being used. Removed.

Reply:
Scratch that, this needs to be all the way up under `ISession` because `RendererBase` only has access to `getSlangSession` as a means of getting the linkage for hashing.

Reply:
Thinking of it now, this can be removed from public interface, and all its logic can be implemented in the new `ComponentType::computeDependencyBasedHash`.

Reply:
The structure of methods should be:
```
 // public interface
IComponentType::computeDependencyBasedHash(MD5Digest** outDigest, int targetIndex, int entryPointIndex);

 // internal abstract method that every subtype of `ComponentType` should implement.
virtual ComponentType::computeDependencyBasedHashImpl(MD5DigestBuilder& builder) = 0;

// implementation of public `IComponentType` interface by simply calling into 
// `ComponentType::computeDependencyBasedHashImpl` + other one-time global things,
// e.g. linkage, target options, entry-point names, etc.
ComponentType::computeDependencyBasedHash(MD5Digest** outDigest, int targetIndex, int entryPointIndex); 
```

Reply:
I am not seeing this being resolved.

Reply:
Fixed locally, will be in next push